src/AppEntry.tsx

```diff
@@ -7,16 +7,23 @@ import { QueryClient, QueryClientProvider } from 'react-query';
 
 const queryClient = new QueryClient();
 
-const AppEntry = () => (
-  <Provider
-    store={init(
-      ...(process.env.NODE_ENV !== 'production' ? [logger] : [])
-    ).getStore()}
-  >
-    <QueryClientProvider client={queryClient}>
-      <App />
-    </QueryClientProvider>
-  </Provider>
-);
+const AppEntry = () => {
+  useEffect(() => {
+    const rootElm = document.getElementById('root');
+    rootElm?.setAttribute('data-ouia-safe', 'true'), [];
+  });
```

Comment:
react 18 removed the callback from the `render` function of `ReactDOM`. this is one of the proposed solutions from the react docs (https://react.dev/blog/2022/03/08/react-18-upgrade-guide, search for "We’ve also removed the callback from render, since it usually does not have the expected result when using Suspense:")

Reply:
@alwayshooin Can you move `[]` for hook dependencies:

```diff
-  rootElm?.setAttribute('data-ouia-safe', 'true'), [];
- });
+  rootElm?.setAttribute('data-ouia-safe', 'true');
+ }, []);
```

Reply:
that's odd that the linter did not catch this or cause rendering issues. 🤔